src/databricks/labs/pytester/fixtures/unwrap.py

```diff
@@ -21,6 +23,20 @@ def call_fixture(fixture_fn: Callable[..., T], *args, **kwargs) -> T:
     return wrapped.obj(*args, **kwargs)
 
 
+_FIXTURES: ContextVar[dict[str, Callable[..., T]]] = ContextVar('fixtures')
+
+
+@contextmanager
+def fixtures(**kwargs: Callable[..., Any]) -> Generator[None, None, None]:
+    prior_fixtures = _FIXTURES.get({})
+    updated_fixtures = {**prior_fixtures, **kwargs}
+    token = _FIXTURES.set(updated_fixtures)
+    try:
+        yield
+    finally:
+        _FIXTURES.reset(token)
```

Comment:
this modifies shared global state and will definitely break when running in multiple threads. why is this necessary? revert.

Reply:
I need this because the default auto-mocks don't allow the double-check waiting inside `make_group()` to complete: the return values of `.create()` and `.list()` on the (account-)groups REST APIs need to be controlled.

At the moment I don't see a test-specific way to pre-configure the mocks that `call_stateful()` will set up as fixtures for the fixture being tested. (I didn't find any other unit tests that need this, but may have missed it or there's another path to the same thing.)

Given that pre-configuring mocks for something under test is a fairly common thing to do, I came up with this. It's specifically intended to not only be thread-safe but also async-safe: I don't understand why it will break when running in multiple threads?

The main alternatives that spring to mind are:

 - Use a specific keyword argument to `call_stateful()` for this purpose that isn't passed through to the fixture under test.
 - Modify the global fixture (inside `CallContext`) that all users of `call_stateful()` have available so that it includes the pre-configured mocks that these group-creation tests use. This would be in the same way as `make_random` (for example) is handled.
 
Do you have a preference, or is there something else I've overlooked?

Reply:
do it without changing a global variable - modify `call_context` right above the `result = ctx[some.__name__](**kwargs)` (the first alternative)

```python
def apply(ctx: CallContext):
    mock_group = Group(id="an_id")
    ctx['ws].groups.create.return_value = mock_group
    ctx['ws].groups.list.return_value = [mock_group]

ctx, group = call_stateful(make_group, call_context_callback=apply)
```